hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -233,28 +237,26 @@ public Table createTable(
     }
 
     private SqlDialect resolveDialect(NodeEngine nodeEngine, String externalDataStoreRef) {
-        try {
-            DataSource dataSource = (DataSource) nodeEngine.getExternalDataStoreService()
-                                                           .getExternalDataStoreFactory(externalDataStoreRef)
-                                                           .getDataStore();
-
-            try (Connection connection = dataSource.getConnection()) {
-
-                SqlDialect dialect = SqlDialectFactoryImpl.INSTANCE.create(connection.getMetaData());
-                String databaseProductName = connection.getMetaData().getDatabaseProductName();
-                switch (databaseProductName) {
-                    case "MySQL":
-                    case "PostgreSQL":
-                    case "H2":
-                        return dialect;
-
-                    default:
-                        LOG.warning("Database " + databaseProductName + " is not officially supported");
-                        return dialect;
-                }
 
+        try (
+                DataStoreHolder<DataSource> dataSource = createDataStore(nodeEngine, externalDataStoreRef);
+                Connection connection = dataSource.get().getConnection()
+        ) {
+
+            SqlDialect dialect = SqlDialectFactoryImpl.INSTANCE.create(connection.getMetaData());
+            String databaseProductName = connection.getMetaData().getDatabaseProductName();
+            switch (databaseProductName) {
```

Comment:
I'd rather have an enum for those constants (additional plus: if we switch to java 17 one day, we will be able to check exhaustiveness) + make comparison ignoring case (e.g. byCode in enum, where by code does ` this.code.equalsIgnoreCase(code)`

Reply:
In general, this could be moved to another place to print the warning only once per datastore factory (currently on each `JdbcSqlConnector#createTable`). I think it needs a rework but in a separate PR

Reply:
see https://github.com/hazelcast/hazelcast/issues/22395